.github/workflows/test-and-lint.yaml

```diff
@@ -21,14 +21,32 @@ jobs:
         with:
           version: 1.3.2
 
-      - name: Install python dependencies
```

Comment:
No need to add this step separately because `Make` notices that the `.venv` is missing and installs the dependencies when we run `make test`.